site/ja/guide/create_op.md

```diff
@@ -0,0 +1,1231 @@
+```
+
+デフォルトでは、テンソルのリストは最小で1の長さを持ちます。
+[対象とするアトリビュートの制約として `">="` ](#デフォルト値と制約) を利用することで、デフォルト値を変更できます。
+次の例では、入力は少なくとも2つの `int32` のテンソルのリストです。
+
+```c++
+REGISTER_OP("MinLengthIntListExample")
+    .Attr("N: int >= 2")
+    .Input("in: N * int32")
+    .Output("out: int32");
+```
+
+同様のシンタックスが、アトリビュート `"list(type)"` に対しても適用できます。
+
+```c++
+REGISTER_OP("MinimumLengthPolymorphicListExample")
+    .Attr("T: list(type) >= 3")
+    .Input("in: T")
+    .Output("out: T");
+```
+
+
+#### 入力と出力
+
+これまでの内容を要約すると、オペレーションの登録では、複数の入力と出力をもつことができます。
+
+```c++
+REGISTER_OP("MultipleInsAndOuts")
+    .Input("y: int32")
+    .Input("z: float")
+    .Output("a: string")
+    .Output("b: int32");
+```
+
+それぞれの入力と出力の指定は、次の形で行います。
+
+```
+<name>: <io-type-expr>
+```
+
+`<name>` は文字で始まり、英数字の文字とアンダースコアから構成できます。
+`<io-type-expr>` は、次の型表現のいずれかになります。
+
+* `<type>`: `<type>` は、サポートする入力型（たとえば、`float`, `int32`, `string`）である。与えられた型の単一のテンソルであることを示す。`tf.DType` を参照のこと。
+
+```c++
+REGISTER_OP("BuiltInTypesExample")
+    .Input("integers: int32")
+    .Input("complex_numbers: complex64");
+```
+
+* `<attr-type>`: `<attr-type>` は、（型の制限がありうる）`type` もしくは `list(type)` の型をもつ [アトリビュート](#アトリビュート) の名前です。このシンタックスは、[ポリモーフィズムなオペレーション](#ポリモーフィズム) を許しています。
+
+```c++
+REGISTER_OP("PolymorphicSingleInput")
+    .Attr("T: type")
+    .Input("in: T");
+
+REGISTER_OP("RestrictedPolymorphicSingleInput")
+    .Attr("T: {int32, int64}")
+    .Input("in: T");
+```
+
+`list(type)` 型のアトリビュートを参照することは、連続したテンソルを受け入れることになります。
+
+```c++
+REGISTER_OP("ArbitraryTensorSequenceExample")
+    .Attr("T: list(type)")
+    .Input("in: T")
+    .Output("out: T");
+
+REGISTER_OP("RestrictedTensorSequenceExample")
+    .Attr("T: list({int32, int64})")
+    .Input("in: T")
+    .Output("out: T");
+```
+
+両方ともおなじ型 `T` であるため、出力 `out` のテンソルの数と型は、入力 `in` とおなじであることに注意してください。
+
+* おなじ型をもつ連続したテンソルについて: `<number> * <type>` となる。`<number>` は、`int` 型である [アトリビュート](#アトリビュート) の名前である。`<type>` は、`tf.DType` または `type` 型のアトリビュートの名前である。最初の例では、オペレーションは `int32` のテンソルのリストを受け取る。
+
+```c++
+REGISTER_OP("Int32SequenceExample")
+    .Attr("NumTensors: int")
+    .Input("in: NumTensors * int32")
+```
+
+一方このオペレーションは、すべておなじである限り、いかなる型のテンソルのリストを受け取ります。
+
+```c++
+REGISTER_OP("SameTypeSequenceExample")
+    .Attr("NumTensors: int")
+    .Attr("T: type")
+    .Input("in: NumTensors * T")
+```
+
+* テンソルのリファレンスついて: `Ref(<type>)` となる。`<type>` は前述した型のいずれかになる。
+
+> 命名に関する注意点: 入力の型として利用されている、いかなるアトリビュートは推測されます。慣例的に、推測されたアトリビュートは、（`T` や `N` のように）大文字の名前を使用します。そうでなければ、入力や出力、アトリビュートは、関数のパラメータ（たとえば `num_outputs`）のような名前を持ちます。詳細は、[前述した命名の注意点](#naming) を参照してください。
+
+詳細は、[`tensorflow/core/framework/op_def_builder.h`][op_def_builder] を参照してください。
+
+
+#### 後方互換性
+
+すばらしいカスタムオペレーションを作り、ほかの人と共有し、顧客がそのオペレーションを使って喜んでいることを想定しましょう。
+しかし、何らかの方法でオペレーションを変更したいとします。
+
+一般的に、既存のチェックインされた仕様の変更は、後方互換でなければなりません。
+すなわち、オペレーションの仕様変更は、前に古い仕様から構築してシリアライズ化された `GraphDef` プロトコルバッファを壊してはいけません。
+`GraphDe` の互換性の詳細は、[ここで説明されています](./versions.md#compatibility_of_graphs_and_checkpoints)。
```

Comment:
[suggestion]
```suggestion
`GraphDef` の互換性の詳細は、[ここで説明されています](./versions.md#compatibility_of_graphs_and_checkpoints)。
```

Reply:
Should we use the full path to point to `versions.md` of the English version?

Reply:
I think we should.
I will check original document and fix it if needs.